crates/ruff_python_ast/src/comparable.rs

```diff
@@ -689,6 +757,103 @@ impl<'a> From<&'a ast::FStringValue> for ComparableFString<'a> {
         }
     }
 }
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct ComparableTString<'a> {
+    elements: Box<[ComparableTStringElement<'a>]>,
+}
+
+impl<'a> From<&'a ast::TStringValue> for ComparableTString<'a> {
+    // The approach below is somewhat complicated, so it may
+    // require some justification.
+    //
+    // Suppose given a t-string of the form
+    // `t"{foo!r} one" " and two " t" and three {bar!s}"`
+    // This decomposes as:
+    // - A `TStringPart::TString`, `t"{foo!r} one"` with elements
+    //      - `TStringElement::Expression` encoding `{foo!r}`
+    //      - `TStringElement::Literal` encoding " one"
+    // - A `TStringPart::Literal` capturing `" and two "`
+    // - A `TStringPart::TString`, `t" and three {bar!s}"` with elements
+    //      - `TStringElement::Literal` encoding " and three "
+    //      - `TStringElement::Expression` encoding `{bar!s}`
+    //
+    // We would like to extract from this a vector of (comparable) t-string
+    // _elements_ which alternate between interpolation elements and literal
+    // elements. In order to do so, we need to concatenate adjacent string
+    // literals. String literals may be separated for two reasons: either
+    // they appear in adjacent string literal parts, or else a string literal
+    // part is adjacent to a string literal _element_ inside of a t-string part.
+    fn from(value: &'a ast::TStringValue) -> Self {
```

Comment:
I think it could be desired to have an `InterpolatedString` enum that abstracts over t and f-strings so that code like this could be reused between. This goes back to my question in the AST crate: What's the motivation for having different nodes (not questioning whether we should have different `FString` and `TString` root nodes but do all nodes have to be different?)